### PR TITLE
feat(test): add read-only smoke tests

### PR DESCRIPTION
**Pull request by anshbansal**
Also fixed `os.putenv` to `os.environ` as the former is for subprocesses created not the current process.

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by RyanHolstien — APPROVED** [review]
Looks good to me!

File: smoke-test/tests/read_only/test_analytics.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        "query": """
+            query getMetadataAnalyticsCharts($input: MetadataAnalyticsInput!) {
+                getMetadataAnalyticsCharts(input: $input) {
+                    groupId
+                    title
+                }
+            }
+        """,
+    }
+    response = frontend_session.post(f"{get_frontend_url()}/api/v2/graphql", json=json)
```
Comment (jjoyce0510) [inline smoke-test/tests/read_only/test_analytics.py:53]: Nice test!

File: smoke-test/tests/read_only/test_search.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            "dataJob",
+        ),
+        (
+            # Pipeline
+            "dataFlow",
+            "dataFlow",
+        ),
+        ("container", "container"),
+        ("tag", "tag"),
+        ("corpUser", "corpUser"),
```
Comment (jjoyce0510) [inline smoke-test/tests/read_only/test_search.py:44]: I'd be surprised if this is working as expected.

Here we need to use the GraphQL notion of EntityType, which is an enum defined here: https://github.com/datahub-project/datahub/blob/master/datahub-graphql-core/src/main/resources/entity.graphql

File: smoke-test/tests/test_result_msg.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+
+datahub_stats = {}
+
+
+def add_datahub_stats(stat_name, stat_val):
+    datahub_stats[stat_name] = stat_val
+
+
+def send_to_slack(passed: str):
```
Comment (jjoyce0510) [inline smoke-test/tests/test_result_msg.py:14]: Very nifty!

File: smoke-test/tests/utils.py [hunk 4]
```diff
@@ -52,14 +52,14 @@ def is_k8s_enabled():
...
-        assert not check_k8s_endpoint(f"{get_gms_url()}/health")
+        assert not check_endpoint(f"{get_frontend_url()}/admin")
+        assert not check_endpoint(f"{get_gms_url()}/health")
     else:
         # Simply assert that docker is healthy, but don't wait.
         assert not check_local_docker_containers()
 
 
-def check_k8s_endpoint(url):
+def check_endpoint(url):
```
Comment (jjoyce0510) [inline smoke-test/tests/utils.py:62]: Nice rename!

**Review by jjoyce0510 — COMMENTED** [review]
Minor comments

**Review by jjoyce0510 — APPROVED** [review]

**Pull request merged**